### PR TITLE
Add gc doc

### PR DESCRIPTION
**Pull request by shiyuhang0**
### Check List 

Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

Code changes

 - Has exported function/method change
 - Has exported variable/fields change
 - Has interface methods change
 - Has persistent data change

Side effects

 - Possible performance regression
 - Increased code complexity
 - Breaking backward compatibility

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation
 - Need to update the `tidb-ansible` repository
 - Need to be included in the release note

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -24,6 +24,26 @@ Most of the TiSpark logic is inside a thin layer, namely, the [tikv-client](http
 - [Dev Guide](https://github.com/pingcap/tispark/wiki/Dev-Guide)
 - [Benchmark](https://github.com/pingcap/tispark/wiki/TiSpark-Benchmark)
 
+## About Garbage Collection
+
+TiKV uses MVCC to control transaction concurrency. When you update the data, the original data is not deleted immediately but is kept together with the new data, with a timestamp to distinguish the version. Garbage Collection (GC) controlled by TiDB will clear the obsolete data before gc safe point periodically.
```
Comment (Daemonxiao) [inline README.md:29]: Please give a link to `MVCC`.

Comment (Daemonxiao) [reply README.md:29]: Also GC.

File: README.md [hunk 2]
```diff
@@ -24,6 +24,26 @@ Most of the TiSpark logic is inside a thin layer, namely, the [tikv-client](http
...
+
+TiSpark guarantee transaction by the distributed reading of the same version. Thus, we need to ensure the version is always behind the gc safe point during the running of the Spark job, or you may get the wrong data.
```
Comment (Daemonxiao) [inline README.md:31]:
```suggestion
TiSpark guarantee transaction by the distributed reading of the same version. Thus, we need to ensure the version is always behind the GC safe point during the running of the Spark job, or you may get the wrong data.
```

File: README.md [hunk 3]
```diff
@@ -24,6 +24,26 @@ Most of the TiSpark logic is inside a thin layer, namely, the [tikv-client](http
...
+
+You can use TiDB system variables [tidb_gc_life_time](https://docs.pingcap.com/tidb/stable/system-variables#tidb_gc_life_time-new-in-v50) to control the gc safe point. The gc safe point will be less than or equal to ${now-tidb_gc_life_time}.
```
Comment (Daemonxiao) [inline README.md:33]:
```suggestion
You can use TiDB system variables [`tidb_gc_life_time`](https://docs.pingcap.com/tidb/stable/system-variables#tidb_gc_life_time-new-in-v50) to control the GC safe point. The GC safe point will be less than or equal to ${now-tidb_gc_life_time}.
```

**Review by Daemonxiao — APPROVED** [review]

**Review by zhangyangyu — APPROVED** [review]

**Review by xuanyu66 — APPROVED** [review]

**Review by zhangyangyu — APPROVED** [review]

**Pull request merged**